pkg/rsstorage/servers/file/server.go

```diff
@@ -127,28 +139,64 @@ func (s *StorageServer) CalculateUsage() (types.Usage, error) {
 
 // diskUsage will walk the specified path in a filesystem and
 // aggregate the size of the contained files.
-func diskUsage(duPath string, cacheTimeout time.Duration) (size datasize.ByteSize, err error) {
-	timeout := time.Now().Add(cacheTimeout)
-	sizep := &size
+func diskUsage(duPath string, cacheTimeout, walkTimeout time.Duration) (size datasize.ByteSize, err error) {
+	stop := make(chan bool)
```

Comment:
That's a good idea. I like using `chan struct{}`, too, when possible. In RSPM, we often use `chan bool` since we need the stop channel to also signal when a service has been successfully stopped. For example, `stop <- true` signals the service to stop, and the the service sends a `stop <- true` signal again when it has shut down.